---
title: "BB 16S MB Final Analyses"
output: html_notebook
---
Load required packages.

```{r}
require(phyloseq)
 require(parallelDist)
require(ggplot2)
  require(randomForest)
require(Boruta)
  require(phyloseq)
require(stats)
  require(dplyr)
require(readr)
  require(pheatmap)
require(tidyverse)
  require(micrUBIfuns)
require(circlize)
  require(phylosmith)
require(phangorn)
  require(gridExtra)
require(pairwiseAdonis)
  require(car)
```

Load necessary files from Dada2 processing script.

```{r}
load("/Users/gordoncuster/Desktop/Git_Projects/ayayee_mouse_2023/Data/CleanedUpDADA2EnvMouse2023_Wtree.RData")
```

Read in metadata

```{#r}
metadata<-read.csv("/Users/gordoncuster/Desktop/Git_Projects/BeeBread/Data/BB_sample-metadata.txt", sep = "\t")
rownames(metadata)<-metadata$X
metadata$Region_Site<-paste( metadata$Region, metadata$Site, sep = "")
metadata$Treatment_Site<-paste( metadata$Treatment, metadata$Site, sep = "")
```

Convert files to correct format for phyloseq object and then rename samples in OTU table to match the format of the metadata file. Create phyloseq object and root phylogenetic tree for usage with distance metrics (e.g., Unifrac).

```{r}
#assign object type to merge into phyloseq object
md<-sample_data(md)
sample_names(md)<-md$SampleID
otu<-otu_table(seqtab, taxa_are_rows = F)
tax_tab<-tax_table(taxa)
sample_names(md)
sample_names(otu)
#check sample names of both otu table and metadata
#remove .fastq.gz from otu table names
sample_names(otu)<-str_split(sample_names(otu), pattern = "_", simplify = T) [,1]
#since they match, you can create the phyloseq object
mouse16S_orig<-phyloseq(md, otu, tax_tab, fitGTR$tree)

#root tree for distance metrics like unifrac
set.seed(11)
phy_tree(mouse16S_orig)<-root(phy_tree(mouse16S_orig), sample(taxa_names(mouse16S_orig), 1), resolve.root = TRUE)
is.rooted(phy_tree(mouse16S_orig))
```

#Pre-processing. 
Rename OTU IDS to something more manageable. We save the ASCV sequences in case we need them later (e.g., BLASTN searches). Create Bacterial and Archaeal phyloseq objects, but first we remove Chloroplasts.

```{r}
#Extract original IDs for future reference. 
seq_df_w_OTUID<-data.frame(OTUID = 1:ntaxa(mouse16S_orig), Sequence = taxa_names(mouse16S_orig))
taxa_names(mouse16S_orig)<-paste("OTU_" , 1:ntaxa(mouse16S_orig), sep = "")


#remove chloroplast
#6 chloroplast or mitochondrial taxa
mouse16S_orig_nc<-subset_taxa(mouse16S_orig, Family!= "Mitochondria" | is.na(Family) & Order!="Chloroplast" | is.na(Order))


# 282 bacterial taxa -1 non bacterial
mouse16S_bac<-subset_taxa(mouse16S_orig_nc, Kingdom == "Bacteria")

```

Examine rarefaction curves. Plateau in most samples under the 1.5k read mark.

```{r}
#rarecurve(otu_table(mouse16S_bac), cex=0.5, main = "Rarefaction Curve", ylab="Richness", xlab="Read Depth")
```

```{r}
set.seed(11)
sort(sample_sums(mouse16S_bac))
summary(sample_sums(mouse16S_bac))
sd(sample_sums(mouse16S_bac))
mouse16S_bac_rarefy<-rarefy_even_depth(mouse16S_bac, sample.size = 1400, rngseed = 14, trimOTUs = T)
#111 otus were removed during rarefaction. 

#To move forward, we can hellinger transform and go from there.
mouse16S_bac_hellinger<-transform_sample_counts(mouse16S_bac, function(x) sqrt(x / sum(x)))

#maximum liklihood point estimates
mouse16S_bac_ML<- transform_sample_counts(mouse16S_bac, function(x) x / sum(x))
```

#Alpha Diversity
```{r}
 plot_richness(mouse16S_bac_rarefy, x = "Groups", measures = c("Shannon", "Observed", "Chao1"), color = "Groups") + geom_boxplot() + ggtitle("Group") + theme_classic()  + xlab("Group")  + theme(axis.text=element_text(size=12, angle = 45, hjust = 0.75),
        axis.title=element_text(size=14))

#sf2 <- grid.arrange(sf2a, sf2b, sf2c, sf2d)
#plot(sf2)

rich_dat <- estimate_richness(mouse16S_bac_rarefy)
rich_dat$Group <- substr(rownames(rich_dat), start = 1, stop = 2) 

kruskal.test(rich_dat$Shannon, rich_dat$Group)
```

```{r}
#create super long color vector
col_vector <- c( "#FFFF00", "#1CE6FF", "#FF34FF", "#FF4A46", "#008941", "#006FA6", "#A30059",
        "#FFDBE5", "#7A4900", "#0000A6", "#63FFAC", "#B79762", "#004D43", "#8FB0FF", "#997D87",
        "#5A0007", "#809693", "#FEFFE6", "#1B4400", "#4FC601", "#3B5DFF", "#4A3B53", "#FF2F80",
        "#61615A", "#BA0900", "#6B7900", "#00C2A0", "#FFAA92", "#FF90C9", "#B903AA", "#D16100",
        "#DDEFFF", "#000035", "#7B4F4B", "#A1C299", "#300018", "#0AA6D8", "#013349", "#00846F",
        "#372101", "#FFB500", "#C2FFED", "#A079BF", "#CC0744", "#C0B9B2", "#C2FF99", "#001E09",
        "#00489C", "#6F0062", "#0CBD66", "#EEC3FF", "#456D75", "#B77B68", "#7A87A1", "#788D66",
        "#885578", "#FAD09F", "#FF8A9A", "#D157A0", "#BEC459", "#456648", "#0086ED", "#886F4C",
        
        "#34362D", "#B4A8BD", "#00A6AA", "#452C2C", "#636375", "#A3C8C9", "#FF913F", "#938A81",
        "#575329", "#00FECF", "#B05B6F", "#8CD0FF", "#3B9700", "#04F757", "#C8A1A1", "#1E6E00",
        "#7900D7", "#A77500", "#6367A9", "#A05837", "#6B002C", "#772600", "#D790FF", "#9B9700",
        "#549E79", "#FFF69F", "#201625", "#72418F", "#BC23FF", "#99ADC0", "#3A2465", "#922329",
        "#5B4534", "#FDE8DC", "#404E55", "#0089A3", "#CB7E98", "#A4E804", "#324E72", "#6A3A4C",
        "#83AB58", "#001C1E", "#D1F7CE", "#004B28", "#C8D0F6", "#A3A489", "#806C66", "#222800",
        "#BF5650", "#E83000", "#66796D", "#DA007C", "#FF1A59", "#8ADBB4", "#1E0200", "#5B4E51",
        "#C895C5", "#320033", "#FF6832", "#66E1D3", "#CFCDAC", "#D0AC94", "#7ED379", "#012C58",
        
        "#7A7BFF", "#D68E01", "#353339", "#78AFA1", "#FEB2C6", "#75797C", "#837393", "#943A4D",
        "#B5F4FF", "#D2DCD5", "#9556BD", "#6A714A", "#001325", "#02525F", "#0AA3F7", "#E98176",
        "#DBD5DD", "#5EBCD1", "#3D4F44", "#7E6405", "#02684E", "#962B75", "#8D8546", "#9695C5",
        "#E773CE", "#D86A78", "#3E89BE", "#CA834E", "#518A87", "#5B113C", "#55813B", "#E704C4",
        "#00005F", "#A97399", "#4B8160", "#59738A", "#FF5DA7", "#F7C9BF", "#643127", "#513A01",
        "#6B94AA", "#51A058", "#A45B02", "#1D1702", "#E20027", "#E7AB63", "#4C6001", "#9C6966",
        "#64547B", "#97979E", "#006A66", "#391406", "#F4D749", "#0045D2", "#006C31", "#DDB6D0",
        "#7C6571", "#9FB2A4", "#00D891", "#15A08A", "#BC65E9", "#FFFFFE", "#C6DC99", "#203B3C",
        "#671190", "#6B3A64", "#F5E1FF", "#FFA0F2", "#CCAA35", "#374527", "#8BB400", "#797868",
        "#C6005A", "#3B000A", "#C86240", "#29607C", "#402334", "#7D5A44", "#CCB87C", "#B88183",
        "#AA5199", "#B5D6C3", "#A38469", "#9F94F0", "#A74571", "#B894A6", "#71BB8C", "#00B433",
        "#789EC9", "#6D80BA", "#953F00", "#5EFF03", "#E4FFFC", "#1BE177", "#BCB1E5", "#76912F",
        "#003109", "#0060CD", "#D20096", "#895563", "#29201D", "#5B3213", "#A76F42", "#89412E",
        "#1A3A2A", "#494B5A", "#A88C85", "#F4ABAA", "#A3F3AB", "#00C6C8", "#EA8B66", "#958A9F",
        "#BDC9D2", "#9FA064", "#BE4700", "#658188", "#83A485", "#453C23", "#47675D", "#3A3F00",
        "#061203", "#DFFB71", "#868E7E", "#98D058", "#6C8F7D", "#D7BFC2", "#3C3E6E", "#D83D66",
        "#2F5D9B", "#6C5E46", "#D25B88", "#5B656C", "#00B57F", "#545C46", "#866097", "#365D25",
        "#252F99", "#00CCFF", "#674E60", "#FC009C", "#92896B")
```


```{r}
plot_bar(mouse16S_bac_rarefy, x = "Sample", y = "Abundance", fill= "Family" )  + scale_fill_manual(values = col_vector)
```

```{r}
rownames(data.frame(otu_table(mouse16S_bac_rarefy)))
colSums(data.frame(otu_table(mouse16S_bac_rarefy)))
```


```{r}
#extract data from phyloseq object
tab_adonis<-data.frame(otu_table(mouse16S_bac_ML))
sd_adonis<-data.frame(sample_data(mouse16S_bac_ML))
sd_adonis$Groups <-as.factor(sd_adonis$Groups)
ord<-metaMDS(tab_adonis)
plot(ord, type = "t")
ord_plot<-plot(ord, type = "t")
ord_plot$species

dist_w<-phyloseq::distance(mouse16S_bac_ML, method =  "wunifrac")
dist_uw<-phyloseq::distance(mouse16S_bac_ML, method =  "unifrac")

#full model with no strata arguemnt. Essentially the effect of each main effect without controlling for any potential groups. 

#weighted
adonis2(dist_w ~  Groups, data= sd_adonis, permutations = 10000)
#unweighted
adonis2(dist_uw ~  Groups, data= sd_adonis, permutations = 10000)

#ordination
ord_nmds<-ordinate(physeq = mouse16S_bac_ML, method = "NMDS", distance = "wunifrac")
ord_nmds$stress
plot_ordination(physeq = mouse16S_bac_ML, ordination = ord_nmds, type = "Samples", color = "Groups") + theme_classic() + ggtitle("Weighted Unifrac NMDS") + theme(plot.title = element_text(hjust = 0.5, size = 18)) + geom_point(size = 3) + theme(axis.text = element_text(size = 14), axis.title=element_text(size=14)) + geom_jitter(position = "jitter")
```

https://github.com/picrust/picrust2/issues/135

https://github.com/picrust/picrust2/issues/136#issuecomment-743696114

https://github.com/picrust/picrust2/wiki/PICRUSt2-Tutorial-(v2.3.0-beta)#minimum-requirements-to-run-full-tutorial












