---
title: "BB 16S MB Final Analyses"
output: html_notebook
---
Load required packages.

```{r}
require(phyloseq)
 require(parallelDist)
require(ggplot2)
  require(randomForest)
require(Boruta)
  require(phyloseq)
require(stats)
  require(dplyr)
require(readr)
  require(pheatmap)
require(tidyverse)
  require(micrUBIfuns)
require(circlize)
  require(phylosmith)
require(phangorn)
  require(gridExtra)
require(pairwiseAdonis)
  require(car)
```

Load necessary files from Dada2 processing script.

```{r}
load("/Users/gordoncuster/Desktop/Git_Projects/ayayee_mouse_2023/Data/CleanedUpDADA2EnvMouse2023_Wtree.RData")
```

Read in metadata

```{#r}
metadata<-read.csv("/Users/gordoncuster/Desktop/Git_Projects/BeeBread/Data/BB_sample-metadata.txt", sep = "\t")
rownames(metadata)<-metadata$X
metadata$Region_Site<-paste( metadata$Region, metadata$Site, sep = "")
metadata$Treatment_Site<-paste( metadata$Treatment, metadata$Site, sep = "")
```

Convert files to correct format for phyloseq object and then rename samples in OTU table to match the format of the metadata file. Create phyloseq object and root phylogenetic tree for usage with distance metrics (e.g., Unifrac).

```{r}
#assign object type to merge into phyloseq object
md<-sample_data(md)
sample_names(md)<-md$SampleID
otu<-otu_table(seqtab, taxa_are_rows = F)
tax_tab<-tax_table(taxa)
sample_names(md)
sample_names(otu)
#check sample names of both otu table and metadata
#remove .fastq.gz from otu table names
sample_names(otu)<-str_split(sample_names(otu), pattern = "_", simplify = T) [,1]
#since they match, you can create the phyloseq object
mouse16S_orig<-phyloseq(md, otu, tax_tab, fitGTR$tree)

#root tree for distance metrics like unifrac
set.seed(11)
phy_tree(mouse16S_orig)<-root(phy_tree(mouse16S_orig), sample(taxa_names(mouse16S_orig), 1), resolve.root = TRUE)
is.rooted(phy_tree(mouse16S_orig))
```

#Pre-processing. 
Rename OTU IDS to something more manageable. We save the ASCV sequences in case we need them later (e.g., BLASTN searches). Create Bacterial and Archaeal phyloseq objects, but first we remove Chloroplasts.

```{r}
#Extract original IDs for future reference. 
seq_df_w_OTUID<-data.frame(OTUID = 1:ntaxa(mouse16S_orig), Sequence = taxa_names(mouse16S_orig))
taxa_names(mouse16S_orig)<-paste("OTU_" , 1:ntaxa(mouse16S_orig), sep = "")


#remove chloroplast
#6 chloroplast or mitochondrial taxa
mouse16S_orig_nc<-subset_taxa(mouse16S_orig, Family!= "Mitochondria" | is.na(Family) & Order!="Chloroplast" | is.na(Order))


# 282 bacterial taxa -1 non bacterial
mouse16S_bac<-subset_taxa(mouse16S_orig_nc, Kingdom == "Bacteria")

```

Examine rarefaction curves. Plateau in most samples under the 1.5k read mark.

```{r}
rarecurve(otu_table(mouse16S_bac), step=50, cex=0.5, main = "Rarefaction Curve", ylab="Richness", xlab="Read Depth")
```

```{r}
set.seed(11)
sort(sample_sums(mouse16S_bac))
summary(sample_sums(mouse16S_bac))
sd(sample_sums(mouse16S_bac))
mouse16S_bac_rarefy<-rarefy_even_depth(mouse16S_bac, sample.size = 1400, rngseed = 14, trimOTUs = T)
#111 otus were removed during rarefaction. 

#To move forward, we can hellinger transform and go from there.
mouse16S_bac_hellinger<-transform_sample_counts(mouse16S_bac, function(x) sqrt(x / sum(x)))

#maximum liklihood point estimates
mouse16S_bac_ML<- transform_sample_counts(mouse16S_bac, function(x) x / sum(x))
```

#Alpha Diversity
```{r}
 plot_richness(mouse16S_bac_rarefy, x = "Groups", measures = c("Shannon", "Observed", "Chao1"), color = "Groups") + geom_boxplot() + ggtitle("Group") + theme_classic()  + xlab("Group")  + theme(axis.text=element_text(size=12, angle = 45, hjust = 0.75),
        axis.title=element_text(size=14))

#sf2 <- grid.arrange(sf2a, sf2b, sf2c, sf2d)
#plot(sf2)
```

```{r}
#create super long color vector
col_vector <- c( "#FFFF00", "#1CE6FF", "#FF34FF", "#FF4A46", "#008941", "#006FA6", "#A30059",
        "#FFDBE5", "#7A4900", "#0000A6", "#63FFAC", "#B79762", "#004D43", "#8FB0FF", "#997D87",
        "#5A0007", "#809693", "#FEFFE6", "#1B4400", "#4FC601", "#3B5DFF", "#4A3B53", "#FF2F80",
        "#61615A", "#BA0900", "#6B7900", "#00C2A0", "#FFAA92", "#FF90C9", "#B903AA", "#D16100",
        "#DDEFFF", "#000035", "#7B4F4B", "#A1C299", "#300018", "#0AA6D8", "#013349", "#00846F",
        "#372101", "#FFB500", "#C2FFED", "#A079BF", "#CC0744", "#C0B9B2", "#C2FF99", "#001E09",
        "#00489C", "#6F0062", "#0CBD66", "#EEC3FF", "#456D75", "#B77B68", "#7A87A1", "#788D66",
        "#885578", "#FAD09F", "#FF8A9A", "#D157A0", "#BEC459", "#456648", "#0086ED", "#886F4C",
        
        "#34362D", "#B4A8BD", "#00A6AA", "#452C2C", "#636375", "#A3C8C9", "#FF913F", "#938A81",
        "#575329", "#00FECF", "#B05B6F", "#8CD0FF", "#3B9700", "#04F757", "#C8A1A1", "#1E6E00",
        "#7900D7", "#A77500", "#6367A9", "#A05837", "#6B002C", "#772600", "#D790FF", "#9B9700",
        "#549E79", "#FFF69F", "#201625", "#72418F", "#BC23FF", "#99ADC0", "#3A2465", "#922329",
        "#5B4534", "#FDE8DC", "#404E55", "#0089A3", "#CB7E98", "#A4E804", "#324E72", "#6A3A4C",
        "#83AB58", "#001C1E", "#D1F7CE", "#004B28", "#C8D0F6", "#A3A489", "#806C66", "#222800",
        "#BF5650", "#E83000", "#66796D", "#DA007C", "#FF1A59", "#8ADBB4", "#1E0200", "#5B4E51",
        "#C895C5", "#320033", "#FF6832", "#66E1D3", "#CFCDAC", "#D0AC94", "#7ED379", "#012C58",
        
        "#7A7BFF", "#D68E01", "#353339", "#78AFA1", "#FEB2C6", "#75797C", "#837393", "#943A4D",
        "#B5F4FF", "#D2DCD5", "#9556BD", "#6A714A", "#001325", "#02525F", "#0AA3F7", "#E98176",
        "#DBD5DD", "#5EBCD1", "#3D4F44", "#7E6405", "#02684E", "#962B75", "#8D8546", "#9695C5",
        "#E773CE", "#D86A78", "#3E89BE", "#CA834E", "#518A87", "#5B113C", "#55813B", "#E704C4",
        "#00005F", "#A97399", "#4B8160", "#59738A", "#FF5DA7", "#F7C9BF", "#643127", "#513A01",
        "#6B94AA", "#51A058", "#A45B02", "#1D1702", "#E20027", "#E7AB63", "#4C6001", "#9C6966",
        "#64547B", "#97979E", "#006A66", "#391406", "#F4D749", "#0045D2", "#006C31", "#DDB6D0",
        "#7C6571", "#9FB2A4", "#00D891", "#15A08A", "#BC65E9", "#FFFFFE", "#C6DC99", "#203B3C",
        "#671190", "#6B3A64", "#F5E1FF", "#FFA0F2", "#CCAA35", "#374527", "#8BB400", "#797868",
        "#C6005A", "#3B000A", "#C86240", "#29607C", "#402334", "#7D5A44", "#CCB87C", "#B88183",
        "#AA5199", "#B5D6C3", "#A38469", "#9F94F0", "#A74571", "#B894A6", "#71BB8C", "#00B433",
        "#789EC9", "#6D80BA", "#953F00", "#5EFF03", "#E4FFFC", "#1BE177", "#BCB1E5", "#76912F",
        "#003109", "#0060CD", "#D20096", "#895563", "#29201D", "#5B3213", "#A76F42", "#89412E",
        "#1A3A2A", "#494B5A", "#A88C85", "#F4ABAA", "#A3F3AB", "#00C6C8", "#EA8B66", "#958A9F",
        "#BDC9D2", "#9FA064", "#BE4700", "#658188", "#83A485", "#453C23", "#47675D", "#3A3F00",
        "#061203", "#DFFB71", "#868E7E", "#98D058", "#6C8F7D", "#D7BFC2", "#3C3E6E", "#D83D66",
        "#2F5D9B", "#6C5E46", "#D25B88", "#5B656C", "#00B57F", "#545C46", "#866097", "#365D25",
        "#252F99", "#00CCFF", "#674E60", "#FC009C", "#92896B")
```


```{r}
plot_bar(mouse16S_bac_rarefy, x = "Sample", y = "Abundance", fill= "Family" )  + scale_fill_manual(values = col_vector)
```

```{r}
rownames(data.frame(otu_table(mouse16S_bac_rarefy)))
colSums(data.frame(otu_table(mouse16S_bac_rarefy)))
```


```{r}
#extract data from phyloseq object
tab_adonis<-data.frame(otu_table(mouse16S_bac_ML))
sd_adonis<-data.frame(sample_data(mouse16S_bac_ML))
sd_adonis$Groups <-as.factor(sd_adonis$Groups)

dist_w<-phyloseq::distance(mouse16S_bac_ML, method =  "wunifrac")
dist_uw<-phyloseq::distance(mouse16S_bac_ML, method =  "unifrac")

#full model with no strata arguemnt. Essentially the effect of each main effect without controlling for any potential groups. 

#weighted
adonis2(dist_w ~  Groups, data= sd_adonis, permutations = 10000)
#unweighted
adonis2(dist_uw ~  Groups, data= sd_adonis, permutations = 10000)

#ordination
ord_nmds<-ordinate(physeq = mouse16S_bac_ML, method = "NMDS", distance = "wunifrac")
ord_nmds$stress
plot_ordination(physeq = mouse16S_bac_ML, ordination = ord_nmds, type = "Samples", color = "Groups") + theme_classic() + ggtitle("Weighted Unifrac NMDS") + theme(plot.title = element_text(hjust = 0.5, size = 18)) + geom_point(size = 3) + theme(axis.text = element_text(size = 14), axis.title=element_text(size=14))
```

https://github.com/picrust/picrust2/issues/135

https://github.com/picrust/picrust2/issues/136#issuecomment-743696114

https://github.com/picrust/picrust2/wiki/PICRUSt2-Tutorial-(v2.3.0-beta)#minimum-requirements-to-run-full-tutorial















Calculate alpha diversity metrics
```{r}
#create alpha diversity table and prep data to include grouping columns
richness_BB16S_bac<-estimate_richness(bb16S_bac_rarefy, measures = c("Shannon", "Observed", "Chao1"))
#add metadata
richness_BB16S_bac$Site<-sample_data(bb16S_bac_rarefy)$Site
richness_BB16S_bac$Timepoint<-sample_data(bb16S_bac_rarefy)$Timepoint
richness_BB16S_bac$Treatment<-sample_data(bb16S_bac_rarefy)$Treatment
richness_BB16S_bac$Treatment_Timepoint<-sample_data(bb16S_bac_rarefy)$Treatment_Timepoint
```

Tests of Shannon diversity 
```{r}
#ANOVA and Tukey's PW comparisons
#Run a full 3 way ANOVA with interactions
mod<-aov(Shannon ~ Site * Treatment * Timepoint, data = richness_BB16S_bac)
#residuals look fine. 
shapiro.test(mod$residuals)
hist(mod$residuals)
summary(mod)
#Summary of model shows site and timepoint significant predictors
#Interaction of site and timepoint

#test of heterogentity 
leveneTest(Shannon ~ Site, data = richness_BB16S_bac)
leveneTest(Shannon ~ Treatment, data = richness_BB16S_bac)
leveneTest(Shannon ~ Timepoint, data = richness_BB16S_bac)

#significant Levene's test for Site. We will run an additional Kruskal Wallis for site. 
kruskal.test(Shannon ~ Site, data = richness_BB16S_bac)

#What about pairwise differences in site
TukeyHSD(mod, "Site")
plot(TukeyHSD(mod, "Site"), las=1, cex.axis = 0.4)

#Several significant pairwise differences to accompany the significant global test for interaction term. We can
TukeyHSD(mod, "Site:Timepoint")
plot(TukeyHSD(mod, "Site:Timepoint"), las=1, cex.axis = 0.4)
```

Tests of Chao 1
```{r}
#ANOVA and Tukey's PW comparisons
#Run a full 3 way ANOVA with interactions
mod<-aov(Chao1 ~ Site * Treatment * Timepoint, data = richness_BB16S_bac)
#residuals look fine. 
shapiro.test(mod$residuals)
hist(mod$residuals)
summary(mod)

leveneTest(Chao1 ~ Site, data = richness_BB16S_bac)
leveneTest(Chao1 ~ Treatment, data = richness_BB16S_bac)
leveneTest(Chao1 ~ Timepoint, data = richness_BB16S_bac)

#Site and the interactions with site are significant. 

TukeyHSD(mod, "Site")
plot(TukeyHSD(mod, "Site"), las=1, cex.axis = 0.4)
```
Tests of Observed - unique taxa in a sample. 
```{r}
#ANOVA and Tukey's PW comparisons
#Run a full 3 way ANOVA with interactions
mod<-aov(Observed ~ Site * Treatment * Timepoint, data = richness_BB16S_bac)
#residuals look fine. 
shapiro.test(mod$residuals)
summary(mod)
hist(mod$residuals)

leveneTest(Observed ~ Site, data = richness_BB16S_bac)
leveneTest(Observed ~ Treatment, data = richness_BB16S_bac)
leveneTest(Observed ~ Timepoint, data = richness_BB16S_bac)

#Summary of model shows site and timepoint to be a significant predictor as well as the site interaction terms.
TukeyHSD(mod, "Site")
plot(TukeyHSD(mod, "Site"), las=1, cex.axis = 0.4)
```

#Beta diversity
```{r}
#extract data from phyloseq object
tab_adonis<-data.frame(otu_table(bb16S_bac_ML))
sd_adonis<-data.frame(sample_data(bb16S_bac_ML))
sd_adonis$Treatment <-as.factor(sd_adonis$Treatment)
sd_adonis$Site <-as.factor(sd_adonis$Site)
dist_w<-phyloseq::distance(bb16S_bac_ML, method =  "wunifrac")
dist_uw<-phyloseq::distance(bb16S_bac_ML, method =  "unifrac")

#full model with no strata arguemnt. Essentially the effect of each main effect without controlling for any potential groups. 

#weighted
adonis2(dist_w ~  Site * Treatment * Timepoint, data= sd_adonis, permutations = 1000)
#unweighted
adonis2(dist_uw ~  Site * Treatment * Timepoint, data= sd_adonis, permutations = 1000)
pairwise.adonis2(dist_uw ~ Treatment + Timepoint, strata = "Site",  data= sd_adonis, nperm = 1000)


#with strata argument
adonis2(dist_w ~  Site * Treatment * Timepoint, data= sd_adonis, strata = sd_adonis$Site, permutations = 1000)
adonis2(dist_uw ~  Site * Treatment * Timepoint, data= sd_adonis, strata = sd_adonis$Site, permutations = 1000)


##PW adonis w/ strata = site
pairwise.adonis2(dist_uw ~ Treatment + Timepoint, strata = "Site",  data= sd_adonis, nperm = 10000)
pairwise.adonis2(dist_w ~ Treatment + Timepoint, strata = "Site",  data= sd_adonis, nperm = 10000)
```
Ordinations
```{r}
#weighted Unifrac
ord_nmds<-ordinate(physeq = bb16S_bac_ML, method = "NMDS", distance = "wunifrac")
ord_nmds$stress
plot_ordination(physeq = bb16S_bac_ML, ordination = ord_nmds, type = "Samples", color = "Site", shape = "Treatment" ) + theme_classic() + ggtitle("Weighted Unifrac NMDS") + theme(plot.title = element_text(hjust = 0.5, size = 18)) + geom_point(size = 3) + theme(axis.text = element_text(size = 14), axis.title=element_text(size=14))
#ggsave("../../Writing/Final_Figs/Weighted_Unifrac_NMDS.eps")
#strongest grouping by Site. Time point also has a smaller effect, but these groupings are more clear than in the CAP ordination. There are several outlier samples that ordinate far to the right on the NMDS1 axis. 

#unweighted Unifrac
ord_nmds<-ordinate(physeq = bb16S_bac_ML, method = "NMDS", distance = "unifrac")
ord_nmds$stress
plot_ordination(physeq = bb16S_bac_ML, ordination = ord_nmds, type = "Samples", color = "Site", shape = "Treatment" ) + theme_classic() + ggtitle("Unweighted Unifrac NMDS") + theme(plot.title = element_text(hjust = 0.5, size = 18)) + geom_point(size = 3) + theme(axis.text = element_text(size = 14), axis.title=element_text(size=14))
#ggsave("../../Writing/Final_Figs/Unweighted_Unifrac_NMDS.eps")


```


Sampling time 2 (post treatment)
```{r}
#post-treatment samples
bb16S_bac_ML_A<-subset_samples(bb16S_bac_ML, Timepoint == "A")

#extract data from phyloseq object for post-treatment
sd_adonis<-data.frame(sample_data(bb16S_bac_ML_A))
sd_adonis$Treatment <-as.factor(sd_adonis$Treatment)
sd_adonis$Site <-as.factor(sd_adonis$Site)
dist_wb<-phyloseq::distance(bb16S_bac_ML_A, method =  "wunifrac")
dist_uwb<-phyloseq::distance(bb16S_bac_ML_A, method =  "unifrac")


#with strata argument
adonis2(dist_wb ~  Site * Treatment, data= sd_adonis, strata = sd_adonis$Site, permutations = 1000)
adonis2(dist_uwb ~  Site * Treatment, data= sd_adonis, strata = sd_adonis$Site, permutations = 1000)

pairwise.adonis2(dist_uwb ~Treatment, strata = "Site", data = sd_adonis, nperm = 1000)

#ordinations
#weighted Unifrac
ord_nmds<-ordinate(physeq = bb16S_bac_ML_A, method = "NMDS", distance = "wunifrac")
ord_nmds$stress
plot_ordination(physeq = bb16S_bac_ML_A, ordination = ord_nmds, type = "Samples", color = "Site", shape = "Treatment" ) + theme_classic() + ggtitle("Weighted Unifrac NMDS") + theme(plot.title = element_text(hjust = 0.5, size = 18)) + geom_point(size = 3)
#ggsave("../../Writing/Final_Figs/Time2_Weighted_Unifrac_NMDS.pdf")


#unweighted Unifrac
ord_nmds<-ordinate(physeq = bb16S_bac_ML_A, method = "NMDS", distance = "unifrac")
ord_nmds$stress
plot_ordination(physeq = bb16S_bac_ML_A, ordination = ord_nmds, type = "Samples", color = "Site", shape = "Treatment" ) + theme_classic() + ggtitle("Unweighted Unifrac NMDS") + theme(plot.title = element_text(hjust = 0.5, size = 18)) + geom_point(size = 3)
#ggsave("../../Writing/Final_Figs/Time2_Unweighted_Unifrac_NMDS.pdf")

```

create taxa bar plot to display makeup
```{r}
sort(table(data.frame(tax_table(bb16S_bac_rarefy))$Phylum), decreasing = TRUE)
length(unique(data.frame(tax_table(bb16S_bac_rarefy))$Phylum))

sort(table(data.frame(tax_table(bb16S_bac_rarefy))$Order), decreasing = TRUE)
length(unique(data.frame(tax_table(bb16S_bac_rarefy))$Order))

sort(table(data.frame(tax_table(bb16S_bac_rarefy))$Family), decreasing = TRUE)
length(unique(data.frame(tax_table(bb16S_bac_rarefy))$Family))

plot_bar(subset_samples(bb16S_bac_rarefy, Timepoint == "A"), fill = "Phylum")  + scale_fill_manual(values = col_vector) + geom_bar(stat = "identity")+ coord_flip()
#ggsave("../../Writing/Final_Figs/taxabarA.pdf")

plot_bar(subset_samples(bb16S_bac_rarefy, Timepoint == "B"), fill = "Phylum")  + scale_fill_manual(values = col_vector) + geom_bar(stat = "identity")+ coord_flip()
#ggsave("../../Writing/Final_Figs/taxabarB.pdf")


```


#Core analysis

```{r}
#create super long color vector
col_vector <- c("#000000", "#FFFF00", "#1CE6FF", "#FF34FF", "#FF4A46", "#008941", "#006FA6", "#A30059",
        "#FFDBE5", "#7A4900", "#0000A6", "#63FFAC", "#B79762", "#004D43", "#8FB0FF", "#997D87",
        "#5A0007", "#809693", "#FEFFE6", "#1B4400", "#4FC601", "#3B5DFF", "#4A3B53", "#FF2F80",
        "#61615A", "#BA0900", "#6B7900", "#00C2A0", "#FFAA92", "#FF90C9", "#B903AA", "#D16100",
        "#DDEFFF", "#000035", "#7B4F4B", "#A1C299", "#300018", "#0AA6D8", "#013349", "#00846F",
        "#372101", "#FFB500", "#C2FFED", "#A079BF", "#CC0744", "#C0B9B2", "#C2FF99", "#001E09",
        "#00489C", "#6F0062", "#0CBD66", "#EEC3FF", "#456D75", "#B77B68", "#7A87A1", "#788D66",
        "#885578", "#FAD09F", "#FF8A9A", "#D157A0", "#BEC459", "#456648", "#0086ED", "#886F4C",
        
        "#34362D", "#B4A8BD", "#00A6AA", "#452C2C", "#636375", "#A3C8C9", "#FF913F", "#938A81",
        "#575329", "#00FECF", "#B05B6F", "#8CD0FF", "#3B9700", "#04F757", "#C8A1A1", "#1E6E00",
        "#7900D7", "#A77500", "#6367A9", "#A05837", "#6B002C", "#772600", "#D790FF", "#9B9700",
        "#549E79", "#FFF69F", "#201625", "#72418F", "#BC23FF", "#99ADC0", "#3A2465", "#922329",
        "#5B4534", "#FDE8DC", "#404E55", "#0089A3", "#CB7E98", "#A4E804", "#324E72", "#6A3A4C",
        "#83AB58", "#001C1E", "#D1F7CE", "#004B28", "#C8D0F6", "#A3A489", "#806C66", "#222800",
        "#BF5650", "#E83000", "#66796D", "#DA007C", "#FF1A59", "#8ADBB4", "#1E0200", "#5B4E51",
        "#C895C5", "#320033", "#FF6832", "#66E1D3", "#CFCDAC", "#D0AC94", "#7ED379", "#012C58",
        
        "#7A7BFF", "#D68E01", "#353339", "#78AFA1", "#FEB2C6", "#75797C", "#837393", "#943A4D",
        "#B5F4FF", "#D2DCD5", "#9556BD", "#6A714A", "#001325", "#02525F", "#0AA3F7", "#E98176",
        "#DBD5DD", "#5EBCD1", "#3D4F44", "#7E6405", "#02684E", "#962B75", "#8D8546", "#9695C5",
        "#E773CE", "#D86A78", "#3E89BE", "#CA834E", "#518A87", "#5B113C", "#55813B", "#E704C4",
        "#00005F", "#A97399", "#4B8160", "#59738A", "#FF5DA7", "#F7C9BF", "#643127", "#513A01",
        "#6B94AA", "#51A058", "#A45B02", "#1D1702", "#E20027", "#E7AB63", "#4C6001", "#9C6966",
        "#64547B", "#97979E", "#006A66", "#391406", "#F4D749", "#0045D2", "#006C31", "#DDB6D0",
        "#7C6571", "#9FB2A4", "#00D891", "#15A08A", "#BC65E9", "#FFFFFE", "#C6DC99", "#203B3C",
        "#671190", "#6B3A64", "#F5E1FF", "#FFA0F2", "#CCAA35", "#374527", "#8BB400", "#797868",
        "#C6005A", "#3B000A", "#C86240", "#29607C", "#402334", "#7D5A44", "#CCB87C", "#B88183",
        "#AA5199", "#B5D6C3", "#A38469", "#9F94F0", "#A74571", "#B894A6", "#71BB8C", "#00B433",
        "#789EC9", "#6D80BA", "#953F00", "#5EFF03", "#E4FFFC", "#1BE177", "#BCB1E5", "#76912F",
        "#003109", "#0060CD", "#D20096", "#895563", "#29201D", "#5B3213", "#A76F42", "#89412E",
        "#1A3A2A", "#494B5A", "#A88C85", "#F4ABAA", "#A3F3AB", "#00C6C8", "#EA8B66", "#958A9F",
        "#BDC9D2", "#9FA064", "#BE4700", "#658188", "#83A485", "#453C23", "#47675D", "#3A3F00",
        "#061203", "#DFFB71", "#868E7E", "#98D058", "#6C8F7D", "#D7BFC2", "#3C3E6E", "#D83D66",
        "#2F5D9B", "#6C5E46", "#D25B88", "#5B656C", "#00B57F", "#545C46", "#866097", "#365D25",
        "#252F99", "#00CCFF", "#674E60", "#FC009C", "#92896B")
```

calculate core at genus and asv level using rarefied data
```{r}
#calculate core at genus level
#325 genera
bb16S_bac_rarefy_tax_genus<-tax_glom(physeq = bb16S_bac_rarefy, taxrank = "Genus", NArm = F)
#calculate core
core_genus<-taxa_core(phyloseq_obj = bb16S_bac_rarefy_tax_genus, treatment = "Site", frequency = 1, abundance_threshold = 0.0001)
core_genus<-taxa_core(phyloseq_obj = core_genus, frequency = 0.5, abundance_threshold = 0.0001)

otu_genus_core <- data.frame(otu_table(core_genus))
tax_genus_core <- data.frame(tax_table(core_genus))
samp_data_genus_core <- data.frame(sample_data(core_genus))

#write.csv(otu_genus_core, "/Users/gordoncuster/Desktop/Git_Projects/BeeBread/Writing/otu_genus_core.csv")
#write.csv(tax_genus_core, "/Users/gordoncuster/Desktop/Git_Projects/BeeBread/Writing/tax_genus_core.csv")
#write.csv(samp_data_genus_core, "/Users/gordoncuster/Desktop/Git_Projects/BeeBread/Writing/samp_data_genus_core.csv")

#table taxonomic info of core genera
table(data.frame(tax_table(core_genus))$Order)
table(data.frame(tax_table(core_genus))$Genus)
#percent of genera assigned to core
14/325
#percent of reads accounted for by the core genera
sum(sample_sums(core_genus))/sum(sample_sums(bb16S_bac_rarefy))

#extract sequence of core genera and export for tree building 
core_genus_names<-rownames(tax_table(core_genus))
core_genera_rep_seq<-seq_df_w_OTUID[seq_df_w_OTUID$OTUID %in% str_split(core_genus_names, pattern = "_", 2, simplify = T)[,2],] 
taxa_df<-data.frame(taxa)
core_genera_tax <- taxa_df[rownames(taxa_df) %in% core_genera_rep_seq$Sequence,]
core_genera_tax$Sequence <- rownames(core_genera_tax)
merged_tax_core_genera_rep_seq<-full_join(core_genera_tax, core_genera_rep_seq)

#write.csv(merged_tax_core_genera_rep_seq, "/Users/gordoncuster/Desktop/Git_Projects/BeeBread/Writing/Core_genera_taxa_rep_seq.csv")
#write.csv(core_genera_rep_seq, "/Users/gordoncuster/Desktop/Git_Projects/BeeBread/Writing/Core_Genera_Rep_Seqs.csv")
#write.csv(core_genera_tax, "/Users/gordoncuster/Desktop/Git_Projects/BeeBread/Writing/Core_Genera_tax.csv")
```

Core Genus
```{r}
#core asv adonis
tab_adonis<-data.frame(otu_table(core_genus))
sd_adonis<-data.frame(sample_data(core_genus))
sd_adonis$Treatment <-as.factor(sd_adonis$Treatment)
sd_adonis$Site <-as.factor(sd_adonis$Site)
dist_core_w_genus<-phyloseq::distance(core_genus, method =  "wunifrac")
dist_core_uw_genus<-phyloseq::distance(core_genus, method =  "unifrac")

#weighted unifrac
adonis2(dist_core_w_genus ~  Site * Treatment * Timepoint, data= sd_adonis, strata = sd_adonis$Site, permutations = 1000)
#unweighted unifrac
adonis2(dist_core_uw_genus ~  Site * Treatment * Timepoint, data= sd_adonis, strata = sd_adonis$Site, permutations = 1000)

#ordinations
#weighted Unifrac
ord_nmds<-ordinate(physeq = core_genus, method = "NMDS", distance = "wunifrac")
ord_nmds$stress
plot_ordination(physeq = core_genus, ordination = ord_nmds, type = "Samples", color = "Site", shape = "Treatment" ) + theme_classic() + ggtitle("Core Genera Weighted Unifrac NMDS") + theme(plot.title = element_text(hjust = 0.5, size = 18)) + geom_point(size = 3)  + theme(axis.text = element_text(size = 14), axis.title=element_text(size=14))
ggsave("../../Writing/Final_Figs/CoreGenus_Weighted_Unifrac_NMDS.eps")
#strongest grouping by Site. Time point also has a smaller effect, but these groupings are more clear than in the CAP ordination. There are several outlier samples that ordinate far to the right on the NMDS1 axis. 

#unweighted Unifrac
ord_nmds<-ordinate(physeq = core_genus, method = "NMDS", distance = "unifrac")
ord_nmds$stress
plot_ordination(physeq = core_genus, ordination = ord_nmds, type = "Samples", color = "Site", shape = "Treatment" ) + theme_classic() + ggtitle("Core Genera Unweighted Unifrac NMDS") + theme(plot.title = element_text(hjust = 0.5, size = 18)) + geom_point(size = 3) + theme(axis.text = element_text(size = 14), axis.title=element_text(size=14))
ggsave("../../Writing/Final_Figs/CoreGenus_Unweighted_Unifrac_NMDS.eps")
```


Core genus at time point 2 
```{r}
core_genus_A <- subset_samples(core_genus, Timepoint == "A")

#core asv adonis
tab_adonis<-data.frame(otu_table(core_genus_A))
sd_adonis<-data.frame(sample_data(core_genus_A))
sd_adonis$Treatment <-as.factor(sd_adonis$Treatment)
sd_adonis$Site <-as.factor(sd_adonis$Site)
dist_core_w_genus<-phyloseq::distance(core_genus_A, method =  "wunifrac")
dist_core_uw_genus<-phyloseq::distance(core_genus_A, method =  "unifrac")

#weighted unifrac
adonis2(dist_core_w_genus ~  Site * Treatment, data= sd_adonis, strata = sd_adonis$Site, permutations = 1000)
#unweighted unifrac
adonis2(dist_core_uw_genus ~  Site * Treatment, data= sd_adonis, strata = sd_adonis$Site, permutations = 1000)
```




Visualization of core microbiome using barplot
```{r}
#pull out core genera names
core_genus_names<-rownames(tax_table(core_genus))
genus_OTU_table<-data.frame(t(otu_table(bb16S_bac_rarefy_tax_genus)))
#check orientation of otu table
rownames(genus_OTU_table)
#add columns corresponding to core status, % sites, % reads
genus_OTU_table$Core <- NA 
for(i in 1:nrow(genus_OTU_table)){
  if(rownames(genus_OTU_table)[i] %in% core_genus_names == T){
  genus_OTU_table[i,]$Core <- "Core"
  } else{
  genus_OTU_table[i,]$Core <- "NonCore"
  }
}

#Prevalence column
genus_OTU_table$prev <- NA
for(i in 1:nrow(genus_OTU_table)){
  genus_OTU_table[i,]$prev<-table(genus_OTU_table[i,(1:69)] > 0)["TRUE"]/ 69}

#proportional abundance
genus_OTU_table$abund <- NA
  for(i in 1:nrow(genus_OTU_table)){
  genus_OTU_table[i,]$abund <- rowSums(genus_OTU_table[i,c(1:69)])/sum(colSums(genus_OTU_table[,c(1:69)]))
  }

dat<-genus_OTU_table[,c(70:72)]
dat$otu <- as.factor(rownames(dat))
dat$otu <- reorder(dat$otu, -dat$prev )
dat$Phylum <- as.factor(data.frame(tax_table(bb16S_bac_rarefy_tax_genus))$Phylum)
dat$number <- 1:nrow(dat)

ggplot(dat, aes(x = otu, y=prev, fill = Phylum)) +
  geom_bar(stat="identity") + scale_fill_manual(values=col_vector) + theme_classic() +
  theme(axis.title.x=element_blank(),
        axis.text.x=element_blank(),
        axis.ticks.x=element_blank()) + ylab("Prevalance") + theme(legend.position = "none")
#ggsave("../../Writing/Final_Figs/core_prev_owlegend.eps")

ggplot(dat, aes(x = otu, y=abund,  fill = Phylum)) +
  geom_bar(stat="identity") + scale_fill_manual(values=col_vector) + theme_classic()  + scale_y_reverse() + theme(axis.text.x = element_text(size = 4 , angle = 45, hjust = 1))
#ggsave("../../Writing/Final_Figs/core_abund.eps")

ggplot(dat, aes(x = otu, y=abund,  fill = Phylum)) +
  geom_bar(stat="identity") + scale_fill_manual(values=col_vector) + theme_classic()  + scale_y_reverse() + theme(axis.text.x = element_text(size = 4 , angle = 45, hjust = 1)) + theme(legend.position = "none") + ylab("Proportional Abundance") + xlab("OTU ID")
#ggsave("../../Writing/Final_Figs/core_abund_wolegend.eps")


#color by core inclusion

ggplot(dat, aes(x = otu, y=prev, fill = Core)) +
  geom_bar(stat="identity") + scale_fill_manual(values=c("#88CCEE", "#999933")) + 
  theme_classic() +
  theme(axis.title.x=element_blank(),
        axis.text.x=element_blank(),
        axis.ticks.x=element_blank()) + ylab("Prevalance") + theme(legend.position = "none")  + theme(axis.text.y = element_text(size=12), axis.title.y = element_text(size = 16))
ggsave("../../Writing/Final_Figs/core_prev_owlegend_Core.eps")

ggplot(dat, aes(x = otu, y=abund,  fill = Core)) +
  geom_bar(stat="identity") + scale_fill_manual(values=c("#88CCEE", "#999933")) + 
  theme_classic()  + scale_y_reverse() + theme(axis.text.x = element_text(size = 4 , angle = 45, hjust = 1, face = "bold"))+ theme( axis.text.x=element_blank(), axis.ticks.x=element_blank())  + theme(axis.text.y = element_text(size=12), axis.title.y = element_text(size = 16))
ggsave("../../Writing/Final_Figs/core_abund_Core.eps")

ggplot(dat, aes(x = otu, y=abund,  fill = Core)) +
  geom_bar(stat="identity") +  scale_fill_manual(values=c("#88CCEE", "#999933")) +
  theme_classic()  + scale_y_reverse() +# theme(axis.text.x = element_text(size = 4 , angle = 45, hjust = 1)) + 
theme(legend.position = "none") + ylab("Proportional Abundance") + xlab("OTU") + theme(axis.text.x=element_blank(), axis.ticks.x=element_blank()) + theme(axis.text.y = element_text(size=12), axis.title.y = element_text(size = 16))
ggsave("../../Writing/Final_Figs/core_abund_wolegend_Core.eps")

```







#### Acidophilic bacteria 
analysis of lactic acid bacteria and acetic acid bacteria at time point 2 (following application)
```{r}
bb16S_bac_ML_A<-subset_samples(bb16S_bac_ML, Timepoint == "A")
#ACETO
#36 taxa
acetobacter <- subset_taxa(bb16S_bac_ML_A, Family =="Acetobacteraceae")
acetobacter_glomed<- tax_glom(acetobacter, taxrank = "Family")

#anova for all taxa combined
dat<-data.frame(otu_table(acetobacter_glomed))
samp_d<-data.frame(sample_data(acetobacter_glomed))
merged_dat<-cbind(dat, samp_d)
#check to make sure md and otu table are in same order. All should be true. 
table(merged_dat$X == rownames(merged_dat))
##mod<-aov(OTU_40 ~ Treatment, data = merged_dat)
#summary(mod)
#plot(residuals(mod))
#boxplot(merged_dat$OTU_40 ~ merged_dat$Treatment)

#baruta for individual taxa
aceto_df <- cbind(data.frame(otu_table(acetobacter)), data.frame(sample_data(acetobacter)))
table(aceto_df$X == rownames(aceto_df))
boxplot(aceto_df$OTU_82 ~ aceto_df$Treatment)
View(tax_table(acetobacter))
#otu 82 ID'ed by baruta. More abundant in ORG. Taxonomy for this taxa "Acetobacteraceae, Bombella"

for(i in 1:length(names(data.frame(otu_table(acetobacter))))){
  boxplot(unlist(aceto_df[i]) ~ aceto_df$Treatment)
}

#very zero inflated so we will not use anova. 
#for(i in 1:length(names(data.frame(otu_table(acetobacter))))){
#  mod<-aov(unlist(aceto_df[i]) ~ aceto_df$Treatment)
#  summary(mod)
#  print(summary(mod))
#  print(names(aceto_df[i]))
#}

for(i in 1:length(names(data.frame(otu_table(acetobacter))))){
  mod<-kruskal.test(unlist(aceto_df[i]) ~ aceto_df$Treatment)
  print(mod)
  print(names(aceto_df[i]))
}


pairwise.wilcox.test(aceto_df$OTU_82, aceto_df$Treatment)
ggplot(aceto_df, aes(x = Treatment, y = OTU_82, color = Treatment)) + geom_boxplot()  + theme_classic()  + theme(axis.text = element_text(size = 14), axis.title=element_text(size=14))
#ggsave("/Users/gordoncuster/Desktop/Git_Projects/BeeBread/Writing/Final_Figs/OTU_282_After_Treatment.eps")

```



```{r}
##LACTO

#75 taxa
lactobac <- subset_taxa(bb16S_bac_ML_A, Order =="Lactobacillales")
lactobac_glomed <- tax_glom(lactobac, taxrank = "Order")

dat<-data.frame(otu_table(lactobac_glomed))
samp_d<-data.frame(sample_data(lactobac_glomed))
merged_dat<-cbind(dat, samp_d)
#check to make sure md and otu table are in same order. All should be true. 
table(merged_dat$X == rownames(merged_dat))

lacto_df <- cbind(data.frame(otu_table(lactobac)), data.frame(sample_data(lactobac)))
table(lacto_df$X == rownames(lacto_df))
boxplot(lacto_df$OTU_530 ~ lacto_df$Treatment)
View(tax_table(lactobac))



for(i in 1:length(names(data.frame(otu_table(lactobac))))){
  boxplot(unlist(lacto_df[i]) ~ lacto_df$Treatment)
}

for(i in 1:length(names(data.frame(otu_table(lactobac))))){
   mod<-kruskal.test(unlist(lacto_df[i]) ~ lacto_df$Treatment)
  print(mod)
  print(names(lacto_df[i]))
}

pairwise.wilcox.test(lacto_df$OTU_568, lacto_df$Treatment)
```




https://stackoverflow.com/questions/61796484/stacked-bar-chart-with-count-above-and-below-x-axis